addons/test_mail/tests/test_message_post.py

```diff
@@ -1058,10 +1058,11 @@ def test_layout_email_lang_context(self):
         self.assertIn('Spanish Layout para', body, 'Layout content should be translated')
         self.assertNotIn('English Layout for', body)
         self.assertIn('Spanish Layout para Spanish description', body, 'Model name should be translated')
-        self.assertIn('SpanishView Spanish description', body, '"View document" should be translated')
-        self.assertNotIn('View %s' % test_records[1]._description, body)
-        self.assertIn('TestSpanishStuff', body, 'Groups-based action names should be translated')
-        self.assertNotIn('TestStuff', body)
+        # python code translation cannot be tested
```

Comment:
Code translation can't be tested in this way.
Our new logic is: if you want a code to be translated in mail, you must add the translation in the PO file of module mail. So, you cannot test code translation for mail in test_mail.
We have some tests for code translation in `test_translation_import`, could you check if it can cover your logic in mail?

Reply:
Actually you have to provider a working fix to this test, especially this is your feature, not ours. You can't break tests and let other people fix it, that's not the way to collaborate between teams :) .

Reply:
OK, I find a workaround. It reuses the 'en.po' in mail to translate and change the translation content.
But I think it actually tests what you want to test for mail :)
https://github.com/odoo/odoo/pull/97692/commits/4cf014ca5c49c7c6098cc1b771699cb3ad99a848